### PR TITLE
rs_log modified so that it can be used by rs_series

### PR DESCRIPTION
**Pull request by Shaswat27**
Modified rs_log so that it handles constant terms, e.g. in rs_log(x+10,x,5). (similar to rs_sin implementation)
Issue #10189

### CODE REVIEW

**Comment by Shaswat27** [comment]
@shivamvats Is this ok?

File: examples/advanced/autowrap_ufuncify.py [hunk 1]
```diff
@@ -24,11 +24,14 @@
 np = import_module('numpy')
 if not np:
     sys.exit("Cannot import numpy. Exiting.")
+plt = import_module('matplotlib.pyplot')
```
Comment (shivamvats) [inline examples/advanced/autowrap_ufuncify.py]: Are changes in this file related to this PR?

**Comment by shivamvats** [comment]
Please have a look at the existing tests and write similar ones for log.

File: examples/advanced/autowrap_ufuncify.py (hunk 1 above)
Comment (Shaswat27) [inline examples/advanced/autowrap_ufuncify.py]: @shivamvats changed by pulling from master when I updated my local repo

**Comment by Shaswat27** [comment]
@shivamvats Will do that and update soon.

File: examples/advanced/autowrap_ufuncify.py (hunk 1 above)
Comment (shivamvats) [inline examples/advanced/autowrap_ufuncify.py]: I don't see any extra commit.
Anyway, you need to properly merge master. There shouldn't be any extra changes.

**Comment by Shaswat27** [comment]
@shivamvats Tests for log have been updated

File: sympy/polys/tests/test_ring_series.py [hunk 2]
```diff
@@ -154,13 +154,31 @@ def test_log():
...
-    p3 = rs_log(p2, x, 9)
-    assert p3 == -p1
+                   7*x**4/36 - x**3/3 + x**2/6 + x
+
+    p = x + 2
+    c = _get_constant_term(p,x)
+    c_ex = c.as_expr()
+    R = R.add_gens([log(c_ex)])
+    p1 = rs_log(x+2,x,4)
+    p2 = p1 - R(log(c_ex))
```
Comment (shivamvats) [inline sympy/polys/tests/test_ring_series.py]: @Shaswat27 You need to write tests for `rs_series`. Something like:
`rs_series(log(x+1), x, 5)`

Moreover, the constant term is supposed to be handled by the code of `rs_log` and not manually.

**Comment by abhiag6891** [comment]
Hey.. Is this issue still open?

**Comment by Shaswat27** [comment]
@abhiag6891 It's still open

**Comment by abhiag6891** [comment]
So if I understand correctly, tests for rs_series need to be written? and also the handling of constant term, right?

**Comment by shivamvats** [comment]
Yes.

**Comment by shivamvats** [comment]
Closing as the issue has been solved by PR #12274.

**Pull request closed without merging**